### PR TITLE
[ios, macos] Fix querying features returning nil when features available.

### LINKED ISSUE #9646: featuresInSourceLayersWithIdentifiers returning empty every time.
**Platform:** IOS
**Mapbox SDK version:** ios-v3.6.0

### Steps to trigger behavior

 1. Fetch the array of features from `MGLVectorSource` by using `sourceLayerIdentifiers` 
 2. It's returning empty array every time.

### Expected behavior

It should return all the features from sourceLayers

### Actual behavior

But it's returning empty array.

Below is my code,

```
func featuresInSourceLayers(WithIdentifiers sourceLayerIdentifiers: [String], andPredicate predicate: NSPredicate?) -> [MGLFeature]{

        if let source = self.mapView.style?.source(withIdentifier: "my-composite") as? MGLVectorSource {
            let features =  source.features(sourceLayerIdentifiers: Set(sourceLayerIdentifiers), predicate: predicate)
            return features
        }
        return [MGLFeature]()
    }
```

I have checked the older SDK `ios-v3.5.4` , it's working there. Thanks.

### PR DESCRIPTION
**Pull request by fabian-guerra**
Fixes #9646

### CODE REVIEW

File: platform/darwin/src/MGLStyle.mm [hunk 1]
```diff
@@ -187,7 +187,13 @@ - (MGLSource *)memberOfSources:(MGLSource *)object {
 - (MGLSource *)sourceWithIdentifier:(NSString *)identifier
 {
     auto rawSource = self.mapView.mbglMap->getSource(identifier.UTF8String);
-    return rawSource ? [self sourceFromMBGLSource:rawSource] : nil;
+    MGLSource *source = nil;
```
Comment (1ec5) [inline platform/darwin/src/MGLStyle.mm]: Nit: it isn’t necessary to explicitly set a local variable to `nil` in Objective-C.

File: platform/darwin/src/MGLStyle.mm [hunk 2]
```diff
@@ -187,7 +187,8 @@ - (MGLSource *)memberOfSources:(MGLSource *)object {
 - (MGLSource *)sourceWithIdentifier:(NSString *)identifier
 {
     auto rawSource = self.mapView.mbglMap->getSource(identifier.UTF8String);
-    return rawSource ? [self sourceFromMBGLSource:rawSource] : nil;
+    
+    return rawSource ? [self sourceFromMBGLSource:rawSource] : nil;;
```
Comment (1ec5) [inline platform/darwin/src/MGLStyle.mm]: Nit: stray semicolon.

File: platform/ios/CHANGELOG.md [hunk 3]
```diff
@@ -7,6 +7,7 @@ Mapbox welcomes participation and contributions from everyone. Please read [CONT
 * Added an MGLStyle.localizesLabels property, off by default, that localizes any Mapbox Streets–sourced symbol layer into the user’s preferred language. ([#9582](https://github.com/mapbox/mapbox-gl-native/pull/9582))
 * Added an additional camera method to MGLMapView that accepts an edge padding parameter. ([#9651](https://github.com/mapbox/mapbox-gl-native/pull/9651))
 * Fixed an issue with the scaling of the user location annotation’s horizontal accuracy indicator. ([#9721](https://github.com/mapbox/mapbox-gl-native/pull/9721))
+* Fixed an issue that caused querying features in a source added through Mapbox Studio returned nil. ([#9784](https://github.com/mapbox/mapbox-gl-native/pull/9784))
```
Comment (1ec5) [inline platform/ios/CHANGELOG.md]: Nit: s/returned/to return/. Also, if I understand correctly, the issue affected all vector sources, not just those added through Mapbox Studio.

Comment (fabian-guerra) [inline platform/ios/CHANGELOG.md]: You are right.

**Comment by 1ec5** [comment]
Made some changes to the changelogs in fdf7a3f3896b62cd4a1041343926af9902d28505:

* The blurb now refers to the specific affected methods, to better distinguish between source querying and layer querying.
* macOS SDK v0.5.1 has yet to be released, because none of the changes that went into iOS SDK v3.6.1 were relevant to macOS.

**Comment by fabian-guerra** [comment]
Thanks @1ec5!

**Pull request merged**